Source/Renderer/ShaderProgram.js

```diff
@@ -27,7 +29,51 @@ define([
     /**
      * @private
      */
+
+    function extractUniforms(shaderText) {
+        // TODO: regex doesn't work for inline structs
+        var uniformNames = [];
+        var uniformLines = shaderText.match(/uniform[\w\s]*(?=[=\[;])/g);
+        if (uniformLines) {
+            for (var i = 0; i < uniformLines.length; i++) {
+                var line = uniformLines[i].trim();
+                var name = line.slice(line.lastIndexOf(' ') + 1);
+                uniformNames.push(name);
+            }
+        }
+        return uniformNames;
+    }
+
     var ShaderProgram = function(options) {
+        // If a uniform exists in both the vertex and fragment shader but with different precision qualifiers,
+        // give the fragment shader uniform a different name. This fixes shader compilation errors on devices
+        // that only support mediump in the fragment shader.
+        var vertexShaderText = options.vertexShaderText;
+        var fragmentShaderText = options.fragmentShaderText;
+        var duplicateUniformNames = {};
+
+        if (!ContextLimits.highpSupported) {
+            var vertexShaderUniforms = extractUniforms(vertexShaderText);
+            var fragmentShaderUniforms = extractUniforms(fragmentShaderText);
+
+            // Find uniforms used in both
+            var i, j;
+            var uniformName;
+            var duplicateName;
+            for (i = 0; i < vertexShaderUniforms.length; i++) {
+                for (j = 0; j < fragmentShaderUniforms.length; j++) {
+                    if (vertexShaderUniforms[i] === fragmentShaderUniforms[j]) {
+                        uniformName = vertexShaderUniforms[i];
+                        duplicateName = uniformName + "_f";
```

Comment:
This, of course, could create a shader that will not compile if a uniform with the same name already exists.  We could test for this and handle it, but, instead, when renaming existing functions and uniforms, we prefix them with `czm_` (for example, see [here](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Scene/Primitive.js#L557)).  In this case, perhaps a `czm_mediump_` prefix is appropriate.

Also, just FYI, GLSL identifiers are limited to 255 characters in WebGL 1; this is not the case in OpenGL.